### PR TITLE
Making 'Add repos' list scrollable

### PR DESCRIPTION
**Pull request by bartaz**
Part of #349

Updated styles of 'Add repos' box to make it scrollable. Leading us one step closer to desired design for this page.

Visual design it's based on:
https://github.com/canonical-websites/build.snapcraft.io/issues/715#issuecomment-302097730
screen shot 2017-06-05 at 16 12 04

Notes:
- currently not responsive (only desktop layout)
- 'Add repos' dialog height is static (not adapting to screen height as defined in spec) (#829)

Implemented:

screen shot 2017-06-05 at 16 12 04

### QA:

1. run local server
2. sign in and go to 'Add repos'
3. wait for repos to load
4. list of repos should be scrollable and have styles as above

### CODE REVIEW

File: src/common/components/private-repos-info/private-repos-info.css [hunk 1]
```diff
@@ -4,7 +4,6 @@ $background-color: #FFF;
 
 .info {
   position: relative;
-  margin-bottom: 24px;
```
Comment (bartaz) [inline src/common/components/private-repos-info/private-repos-info.css]: Bottom margin not needed anymore due to style changes.

File: src/common/components/select-repositories-page/index.js [hunk 2]
```diff
@@ -32,13 +31,15 @@ class SelectRepositoriesPage extends Component {
     return (
       <div>
         <FirstTimeHeading />
-        <CardHighlighted>
-          <HeadingThree className={ styles.heading }>
-            Choose repos to add
-          </HeadingThree>
-          <PrivateReposInfo user={ this.props.user } onRefreshClick={this.onRefresh.bind(this)}/>
+        <div className={styles.selectRepositoriesBox}>
```
Comment (bartaz) [inline src/common/components/select-repositories-page/index.js]: Replaced `CardHighlighted` with custom styling.

File: src/common/components/select-repositories-page/select-repositories-page.css [hunk 3]
```diff
@@ -1,3 +1,12 @@
 .heading {
   margin-bottom: 12px;
 }
+
+.selectRepositoriesBox {
```
Comment (bartaz) [inline src/common/components/select-repositories-page/select-repositories-page.css]: Styles similar to Vanilla card, but without padding.

File: src/common/components/select-repository-list/index.js [hunk 4]
```diff
@@ -107,15 +106,19 @@ export class SelectRepositoryListComponent extends Component {
...
+            <div className={ styles.spinnerWrapper }>
+              <div className={ spinnerStyles }><Spinner /></div>
+            </div>
+          }
+          { renderedRepos }
+          { pagination }
+        </div>
         <div className={ styles.footer }>
-          <HeadingThree>
+          <strong>
```
Comment (bartaz) [inline src/common/components/select-repository-list/index.js]: Selected repos should not be a heading, just regular text in bold.

File: src/common/components/select-repository-list/index.js [hunk 5]
```diff
@@ -136,7 +139,9 @@ export class SelectRepositoryListComponent extends Component {
   }
 
   renderPageLinks(pageLinks) {
-    if (pageLinks) {
+    const hasPagination = !!(pageLinks && (pageLinks.first || pageLinks.last));
```
Comment (bartaz) [inline src/common/components/select-repository-list/index.js]: Correctly checking if pagination should be visible - only if we have any paging links.

File: src/common/components/select-repository-list/styles.css [hunk 6]
```diff
@@ -5,7 +5,20 @@
...
+  padding: 1rem;
   display: flex;
   justify-content: space-between;
-  margin: 60px 0;
+  align-items: baseline;
+}
+
+.repoList {
+  height: 300px;
+  overflow: auto;
```
Comment (bartaz) [inline src/common/components/select-repository-list/styles.css]: Making repos list scrollable in a fixed height of 300px.

Height of the list should depend on viewport size, but it's to be implemented separately.

File: src/common/components/private-repos-info/private-repos-info.css [hunk 7]
```diff
@@ -4,7 +4,6 @@ $background-color: #FFF;
...
 }
 
 .info ul {
```
Comment (richmccartney) [inline src/common/components/private-repos-info/private-repos-info.css]: Can we change the ```<ul>``` to be a class? With current styling in Vanilla we try to not style on the base elements but effect them with classes.

**Comment by richmccartney** [comment]
@bartaz Made one comment, I didn't find any other issues 👍

**Pull request merged**